react-native-aztec/src/AztecView.js

```diff
@@ -91,59 +87,65 @@ class AztecView extends React.Component {
   }
 
   _onFocus = (event) => {
-    if (!this.props.onFocus) {
-      return;
-    }
-
+    this.focused = true;
     const { onFocus } = this.props;
-    onFocus(event);
+
+    if (onFocus) {
+      onFocus(event);
+    }
   }
   
   _onBlur = (event) => {
+    this.focused = false;
     this.selectionEndCaretY = null;
-    TextInputState.blurTextInput(ReactNative.findNodeHandle(this));
+    const { onBlur } = this.props;
 
-    if (!this.props.onBlur) {
-      return;
+    if (onBlur) {
+      onBlur(event);
     }
-
-    const { onBlur } = this.props;
-    onBlur(event);
   }
 
   _onSelectionChange = (event) => {
-    if ( this.props.onSelectionChange ) {
+    if (this.focused && this.props.onSelectionChange) {
       const { selectionStart, selectionEnd, text } = event.nativeEvent;
       const { onSelectionChange } = this.props;
       onSelectionChange( selectionStart, selectionEnd, text, event );
     }
 
-    if ( this.props.onCaretVerticalPositionChange && 
-      this.selectionEndCaretY != event.nativeEvent.selectionEndCaretY ) {
+    if (this.focused && this.props.onCaretVerticalPositionChange && 
+      this.selectionEndCaretY != event.nativeEvent.selectionEndCaretY) {
         const caretY = event.nativeEvent.selectionEndCaretY;
         this.props.onCaretVerticalPositionChange( event.target, caretY, this.selectionEndCaretY );
         this.selectionEndCaretY = caretY;
     }
   }
 
   blur = () => {
-    TextInputState.blurTextInput(ReactNative.findNodeHandle(this));
+    if (this.focused) {
+      if (Platform.OS === "ios") {
+        UIManager.blur(ReactNative.findNodeHandle(this))
+      } else {
+        this.dispatch(UIManager.getViewManagerConfig('AndroidTextInput').Commands.blurTextInput)
+      }
+    }
+    this.focused = false;
   }
 
   focus = () => {
-    TextInputState.focusTextInput(ReactNative.findNodeHandle(this));
+    if (!this.focused) {
```

Comment:
Could you help me understand why we need to change here? how this is giving a different result than the check TextInputState uses? 

```
function focusTextInput(textFieldID: ?number) {
  if (currentlyFocusedID !== textFieldID && textFieldID !== null) {
  ...//focus
  }
}
```